packages/utils/typescript/lib/utils/get-config-path.js

```diff
@@ -16,7 +16,7 @@ const DEFAULT_TS_CONFIG_FILENAME = 'tsconfig.json';
  * @return {string | undefined}
  */
 module.exports = (dir, { filename = DEFAULT_TS_CONFIG_FILENAME, ancestorsLookup = false } = {}) => {
-  const dirAbsolutePath = path.resolve(dir);
+  const dirAbsolutePath = path.resolve(dir).split(path.sep).join(path.posix.sep);
```

Comment:
Maybe I'm missing something, but I'm not sure to understand the need here. Isn't `path.resolve` supposed to return the path with os path separator anyway?

Reply:
It does return it using the platform-specific path segment separator, correct.
This is why we need to manually fix this, as the result will be different [depending on the OS](https://nodejs.org/api/path.html#pathsep) we are on:
- `\` on Windows
- `/` on POSIX

`ts.findConfigFile()` on line 20 always returns the path with POSIX style path separators (e.g. "foo/bar").
When we then compare it to the Windows style path that `path.resolve()` returns using `.startsWith()` in line 26, we are comparing "foo/bar" with "foo\\bar" which will always return `false`.

This is what my change is doing, it ensures that the path is always using the POSIX style path separators, even on Windows. This is why this only fails on Windows and was most likely overlooked during development of this change.

Reply:
This is the correct fix.
I would also add ".split(path.sep).join(path.posix.sep);" to configFilePath just in case it changes in typescript in the future.

Reply:
Thanks for the clarifications.

I'm now just wondering what happens if we try to compare the configFilePath (always posix) to the dirAbsolutePath (sep bound to the OS) on line 26. Shouldn't we do the check on two posix paths instead?

Reply:
Basically for configFilePath.startsWith(dirAbsolutePath) with configFilePath as a / separated path and dirAbsolutePath with separators that depends on the OS, we can have mismatches, right?

Reply:
The issue you describe is what is solved with the proposed change.

The path seperator of `dirAbsolutePath` was dependent on the OS of the host before, but by splitting it by the OS sep and joining with the POSIX sep (`.split(path.sep).join(path.posix.sep)`) we convert it into POSIX format, thus now comparing POSIX with POSIX style separators in line 26.

I didn't want to touch any other logic so this was the only change I made.

Reply:
@Convly What's holding us back from getting this merged? I feel like this should be fixed ASAP as it's preventing Windows users from starting any TS Strapi projects.